service/controller/resource/deployment/update.go

```diff
@@ -73,6 +79,47 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		return nil, microerror.Mask(err)
 	}
 
+	// Create a client for the reconciled tenant cluster
+	var tcK8sClient kubernetes.Interface
+	{
+		if r.tenantCluster != nil {
+			customObject, err := key.ToCustomObject(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			r.logger.LogCtx(ctx, "level", "debug", "message", "creating Kubernetes client for tenant cluster")
+
+			i := key.ClusterID(customObject)
+			e := key.ClusterAPIEndpoint(customObject)
+
+			restConfig, err := r.tenantCluster.NewRestConfig(ctx, i, e)
+			if tenantcluster.IsTimeout(err) {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "did not create Kubernetes client for tenant cluster")
+				r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for certificates timed out")
+			} else if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			clientsConfig := k8sclient.ClientsConfig{
+				Logger:     r.logger,
+				RestConfig: restConfig,
+			}
+			k8sClients, err := k8sclient.NewClients(clientsConfig)
+			if tenant.IsAPINotAvailable(err) {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available")
+				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			} else if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			tcK8sClient = k8sClients.K8sClient()
+			r.logger.LogCtx(ctx, "level", "debug", "message", "created Kubernetes client for tenant cluster")
+		} else {
+			r.logger.LogCtx(ctx, "level", "warning", "message", "unable to create Kubernetes client for tenant cluster")
+			tcK8sClient = nil
+		}
+	}
+
```

Comment:
Easiest fix for your issue:
Move code from line `124` until `188` into a new private method which has a signature like
```
func (r *Resource) someFunc(ctx context.Context, obj  interface{}, currentDeployments []v1.Deployments, desiredDeployments []v1.Deployments, tcK8sClient kubernetes.Interface ) ([]*v1.Deployment, error)
```
And point the existing unit tests at `someFunc`